### ISSUE TITLE
Ubnormal达不到文章数据

### ISSUE THREAD

**Issue by kj500-w** [guijiejie/DCMD-main#2]
Hi, thanks for releasing the DCMD code.

I am reproducing the UBnormal results reported in the paper.
For UBnormal, the paper reports 69.0 AUC for both HR-UBnormal and UBnormal, but I get around 64 AUC.

My setting:
- Dataset: UBnormal
- n_epochs: 100
- batch_size: 1024
- noise_steps: 10
- n_generated_samples: 50
- loss_1_series_weight: 0.01
- loss_2_prior_weight: 0.01
- I tried both:
  1. lr=1e-4, StepLR(step=36, gamma=0.5)
  2. lr=1e-3, ExponentialLR(gamma=0.99)

Results:
- lr=1e-4:
  - UBnormal: about 64.07 AUC
  - HR-UBnormal: about 64.71 AUC
- lr=1e-3 + ExponentialLR:
  - UBnormal: about 63.77 AUC
  - HR-UBnormal: about 64.41 AUC

I also noticed that the released UBnormal test config uses:
aggregation_strategy: all

However, running evaluation with aggregation_strategy=all causes a shape mismatch error in post_processing / compute_var_matrix, because the output loss has shape [B, 50] instead of [B].

Could you please clarify:
1. What exact training config was used for the reported UBnormal / HR-UBnormal 69.0 AUC?
2. Which aggregation_strategy should be used for UBnormal evaluation?
3. Is aggregation_strategy=all expected to work with the released post_processing code?
4. What are the expected numbers of loaded / remaining UBnormal test trajectories after removing short trajectories?
5. Are the HR-UBnormal and UBnormal results evaluated from the same checkpoint?

**Comment by hongsong-wang** [comment]
Thank you for your question. The master's student has already graduated and is currently occupied with work, so a timely response may not be possible.

The evaluation aggregation strategy is mean.  For dimension errors, you can try compressing the [B, 50] output from all by taking the mean to obtain a one-dimensional vector. HR-UBnormal and UBnormal are independently trained checkpoints.

**Comment by kj500-w** [comment]
Thank you very much for the clarification.

I have several follow-up questions about reproducing the UBnormal / HR-UBnormal results.

1. About aggregation_strategy=all

You mentioned that the [B, 50] output should be compressed by taking the mean to obtain a one-dimensional vector.

Could you please confirm whether the mean is applied directly on the [B, 50] loss tensor before post-processing, i.e.,

    out = mean(out, dim=1)

or whether the 50 generated samples are first converted to frame-level anomaly scores separately and then averaged at the frame-score level?

In my implementation, I apply mean over the [B, 50] loss tensor before compute_var_matrix.

2. About independent checkpoints for UBnormal and HR-UBnormal

You mentioned that HR-UBnormal and UBnormal use independently trained checkpoints.

Could you please clarify how the HR-UBnormal checkpoint was trained?

- Was it trained with the same UBnormal training data but using a separate random initialization / training run?
- Or was the training data filtered using the HR masks so that only human-related frames or clips were used during training?

In the released code, use_hr seems to affect only evaluation post-processing, not training data loading.

3. About the exact training configuration

For the reported UBnormal and HR-UBnormal results, could you please confirm the exact training hyperparameters?

- learning rate: 1e-4 or 1e-3?
- scheduler: StepLR or ExponentialLR?
- scheduler gamma?
- number of epochs?
- batch size?
- loss_1_series_weight and loss_2_prior_weight?

The paper mentions 1e-4, while the released UBnormal config uses 1e-3, so I want to make sure which one was used for the reported results.

4. About checkpoint selection / monitored metric

Since the released UBnormal training config has validation=false, I would like to ask how the final checkpoint was selected.

- Which metric was monitored for saving checkpoints: "loss", "loss1", validation AUC, or another metric?
- Were only the saved top-k checkpoints evaluated?
- Or were checkpoints from all epochs evaluated and the best test AUC reported?

I noticed that train loss may not correlate well with the final AUC on UBnormal.

5. About expected data statistics

Could you please provide the expected number of loaded trajectories and remaining trajectories after removing short trajectories for UBnormal?

For example, in my evaluation log for UBnormal test, I get:

    Loaded 2964 trajectories.
    Removed short trajectories. Number of trajectories left: 2216.

Could you please confirm whether these numbers are consistent with your preprocessing?

It would also be helpful to know the expected numbers for:
- UBnormal training split
- UBnormal test split
- HR-UBnormal training/evaluation if different

6. About post-processing parameters

For the reported UBnormal and HR-UBnormal results, did you use the following values exactly?

    filter_kernel_size = 30
    frames_shift = 18
    n_generated_samples = 50

Thank you again for your help. These details would be very useful for reproducing the reported UBnormal / HR-UBnormal results.
